receiver/snmpreceiver/config.go

```diff
@@ -0,0 +1,592 @@
+	errEmptyPrivacyType     = errors.New("privacy_type must be specified when security_level is auth_priv")
+	errBadPrivacyType       = errors.New("privacy_type must be either DES, AES, AES192, AES192C, AES256, AES256C")
+	errEmptyPrivacyPassword = errors.New("privacy_password must be specified when security_level is auth_priv")
+	errMetricRequired       = errors.New("must have at least one config under metrics")
+)
+
+// Config defines the configuration for the various elements of the receiver agent.
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+
+	// The SNMP target to send data to. Must be formatted as udp/tcp[4/6]://{host}:{port}. Default: udp://localhost:161
+	Endpoint string `mapstructure:"endpoint"`
+
+	// The version of SNMP. Valid options: v1, v2c, v3. Default: v2c
+	Version string `mapstructure:"version"`
+
+	// The SNMP community string to use. Default: public
+	Community string `mapstructure:"community"`
+
+	// Only valid for version “v3”
+	User string `mapstructure:"user"`
+
+	// Only valid for version “v3”
+	// Valid options: “no_auth_no_priv”, “auth_no_priv”, “auth_priv”
+	// Default: "no_auth_no_priv"
+	SecurityLevel string `mapstructure:"security_level"`
+
+	// Only valid for version “v3” and if “no_auth_no_priv” is not selected for security_level
+	// Valid options: “md5”, “sha”, “sha224”, “sha256”, “sha384”, “sha512”
+	AuthType string `mapstructure:"auth_type"`
+
+	// Only valid if authentication_type specified
+	AuthPassword string `mapstructure:"auth_password"`
+
+	// Only valid for version “v3” and if “auth_priv” selected for security_level
+	// Valid options: “des”, “aes”, “aes192”, “aes256”, “aes192c”, “aes256c”
+	PrivacyType string `mapstructure:"privacy_type"`
+
+	// Only valid if privacy_type specified
+	PrivacyPassword string `mapstructure:"privacy_password"`
+
+	// Resource attributes. The attribute names are used for the keys
+	ResourceAttributes map[string]ResourceAttributeConfig `mapstructure:"resource_attributes"`
+
+	// Attributes. The attribute names are used for the keys
+	Attributes map[string]AttributeConfig `mapstructure:"attributes"`
+
+	// Metrics. The metric names are used for the keys
+	Metrics map[string]MetricConfig `mapstructure:"metrics"`
+}
+
+// ResourceAttributeConfig defines
+type ResourceAttributeConfig struct {
+	Description string `mapstructure:"description"`
+	// An optional value used to map returned values from a table column OID to a resource attribute values
+	// This OID must be related to the same table that is used to match a different column OID a defined metric
+	// In addition, the resource attribute that this OID belongs to must be assigned to the same metric
+	// If OID is not used, then IndexedValuePrefix must be used instead
+	OID string `mapstructure:"oid"` // Used to assign return values from one or more indexed OIDs to a resource attribute using
+	// a new indexed OID within the same table as one or more metric indexed OIDs
```

Comment:
nit: These comments seem odd in the fact they are following the field and on the line under it.

Reply:
@cpheps config.go is the one file that i still need to revisit the comments.